### PR TITLE
[docs-java-client] Client V2 Configuration Docs 

### PR DESCRIPTION
**Pull request by chernser**
## Summary
- Added list of all JAVA Client V2 configuration parameters 
- Added documentation for authentication method.

## Checklist
- [ ] Delete items not relevant to your PR
- [ ] URL changes should add a redirect to the old URL via https://github.com/ClickHouse/clickhouse-docs/blob/main/docusaurus.config.js
- [ ] If adding a new integration page, also add an entry to the integrations list here: https://github.com/ClickHouse/clickhouse-docs/blob/main/docs/en/integrations/index.mdx

### CODE REVIEW

File: docs/en/integrations/language-clients/java/client-v2.md [hunk 1]
```diff
@@ -71,6 +113,63 @@ Major configuration parameters are defined in one scope (client or operation) an
...
+| httpHeader(String key, Collection values) | - `key` - HTTP header key.<br /> - `values` - list of string values. | Sets values for a single HTTP header. Previous value is overriden.|
+| httpHeaders(Map headers) | - `header` - map with HTTP headers and their values. | Sets multiple HTTP header values at a time. |
+| serverSetting(String name, String value) | - `name` - name of a query level setting.<br /> - `value` - string value of the setting. | Sets query level setting to be sent along with every request. Operation settings may override it. See [Query Level Settings](/docs/en/operations/settings/query-level) for more information. | 
+| serverSetting(String name,  Collection values) | - `name` - name of a query level setting.<br /> - `values` - string values of the setting. | Sets query level setting values to be sent along with every request. Operation settings may be override it. This method is useful to set settings with multiple values, for example [roles](/docs/en/interfaces/http#setting-role-with-query-parameters) |
```
Comment (mshustov) [inline docs/en/integrations/language-clients/java/client-v2.md:168]: > name of a query level setting.

Does it set the server setting? I'd rephrase it something like `clickhouse setting set for each query`

File: docs/en/integrations/language-clients/java/client-v2.md [hunk 2]
```diff
@@ -71,6 +113,63 @@ Major configuration parameters are defined in one scope (client or operation) an
...
+| useAsyncRequests(boolean async) | `async` - flag that indicates if the option should be enabled. | Sets if client should execute request in a separate thread. Disabled by default because application knows better how to organize multithreaded tasks and running tasks in separate thread do not help with performance. | 
+| setSharedOperationExecutor(ExecutorService executorService) | `executorService` - instance of executor service. | Sets executor service for operation tasks. | 
+| setClientNetworkBufferSize(int size) | - `size` - size in bytes | Sets size of a buffer in application memory space that is used to copy data back-and-forth between socket and application. Greater reduces system calls to TCP stack, but affects how much memory is spent on every connection. This buffer is also subject for GC because connections are shortlive. Also keep in mind that allocating big continious block of memory might be a problem. |
+| retryOnFailures(ClientFaultCause ...causes) | - `causes` - array of causes that causes retry | Set if client should retry on certain faults. This option is useful to avoid some infrequent problems like staled connection. |
```
Comment (mshustov) [inline docs/en/integrations/language-clients/java/client-v2.md:161]: it's unclear how `causes` looks like. can we add a link to an example?

File: docs/en/integrations/language-clients/java/client-v2.md [hunk 3]
```diff
@@ -71,6 +113,63 @@ Major configuration parameters are defined in one scope (client or operation) an
...
 
+## Client Configuration
+
+| Configuration Method                  | Arguments                                      |  Description                                |
+|---------------------------------------|:-----------------------------------------------|:--------------------------------------------|
+| addEndpoint(String endpoint)          | - `enpoint` - URL formatted a server address.      | Adds a server endpoint to list of available servers. Currently only one endpoint is supported. |
+| addEndpoint(Protocol protocol, String host, int port, boolean secure) | - `protocol` - connection protocol `com.clickhouse.client.api.enums.Protocol#HTTP`.<br />- `host` - IP or hostname of a server.<br />- `secure` - if communication should use secure version of the protocol (HTTPs) | Adds a server endpoint to list of available servers. Currently only one endpoint is supported. |
+| setOption(String key, String value)   | - `key` - String key of an option.<br /> - `value` - String value of an option | Sets raw value of client options. Useful when reading configuration from properties files. It helps to avoid calling corresponding builder methods. |
```
Comment (mshustov) [inline docs/en/integrations/language-clients/java/client-v2.md:122]: How about
> `key` - String key of a client configuration option. 

should add a link to the list of client configuration options?

**Review by mshustov — APPROVED** [review]

File: docs/en/integrations/language-clients/java/client-v2.md (hunk 3 above)
Comment (chernser) [reply docs/en/integrations/language-clients/java/client-v2.md:122]: Sounds good, but we do not have such list yet. Will work on it in 0.7.2 release.
